File: src/main/java/com/github/stars/controller/AiAnalyzeController.java
package com.github.stars.controller;

import com.github.stars.service.AiAnalyzeService;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import java.util.*;

@RestController
@RequestMapping("/api/analyze")
public class AiAnalyzeController {

    @Resource
    private AiAnalyzeService aiAnalyzeService;

    /**
     * 启动 AI 分析任务（异步）
     */
    @PostMapping("/start")
    public Map<String, Object> startAnalyze(
            @RequestParam(value = "keyword", defaultValue = "") String keyword,
            @RequestParam(value = "language", defaultValue = "") String language,
            @RequestParam(value = "categoryIds", defaultValue = "") String categoryIds,
            @RequestParam(value = "sortBy", defaultValue = "starred_at") String sortBy,
            @RequestParam(value = "sortOrder", defaultValue = "desc") String sortOrder) {

        Map<String, Object> result = new LinkedHashMap<>();
        String taskId = aiAnalyzeService.createAnalyzeTask(keyword, language, categoryIds, sortBy, sortOrder);
        result.put("success", true);
        result.put("taskId", taskId);
        result.put("message", "AI 分析任务已启动（最多分析 30 个项目）");
        return result;
    }

    /**
     * 查询分析任务状态和结果
     */
    @GetMapping("/task/{taskId}")
    public Map<String, Object> getTaskStatus(@PathVariable String taskId) {
        return aiAnalyzeService.getTaskStatus(taskId);
    }
}

File: backend-nestjs/src/ai/controllers/ai-analyze.controller.ts
import { Controller, Post, Get, Param, Query } from '@nestjs/common';
import { AiAnalyzeService } from '../services/ai-analyze.service';

@Controller('api/analyze')
export class AiAnalyzeController {
    constructor(private readonly service: AiAnalyzeService) {}

    /**
     * 启动 AI 分析任务
     *
     * @param q.keyword 关键词筛选（可选）
     * @param q.language 语言筛选（可选）
     * @param q.categoryIds 分类 ID 筛选（可选）
     * @param q.sortBy 排序字段（默认 starred_at）
     * @param q.sortOrder 排序方向（默认 desc）
     * @returns 任务 ID 和状态信息
     */
    @Post('start')
    async start(@Query() q: any) {
        const taskId = this.service.createAnalyzeTask(
            q.keyword || '',
            q.language || '',
            q.categoryIds || '',
            q.sortBy || 'starred_at',
            q.sortOrder || 'desc',
        );
        return { success: true, taskId, message: 'AI分析任务已启动' };
    }

    /**
     * 查询分析任务状态
     *
     * @param taskId 任务 ID
     * @returns 任务状态信息（status、content 等）
     */
    @Get('task/:taskId')
    async task(@Param('taskId') taskId: string) {
        return this.service.getTaskStatus(taskId);
    }
}
